From 33c172a3f2c6ee5ed7594aedf84f0fa950ceaa6a Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <jkysela@redhat.com>
Date: Wed, 3 Sep 2014 07:35:11 -0400
Subject: [alsa] core: Use ktime_get_ts()

Message-id: <140972823573-336-patchset-jkysela@redhat.com>
Patchwork-id: 91032
O-Subject: [RHEL 7.1 PATCH 335/442] ALSA: core: Use ktime_get_ts()
Bugzilla: 1112200
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>

do_posix_clock_monotonic_gettime() is a leftover from the initial
posix timer implementation which maps to ktime_get_ts().

Signed-off-by: Thomas Gleixner <tglx@linutronix.de>
Signed-off-by: Takashi Iwai <tiwai@suse.de>

(picked from kernel: 26204e048d2ee0c65e0539f7cc2b66f845a19a41)
rhbz#1112200
Signed-off-by: Jaroslav Kysela <jkysela@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/include/sound/pcm.h b/include/sound/pcm.h
index b4d6697..d854fb3 100644
--- a/include/sound/pcm.h
+++ b/include/sound/pcm.h
@@ -932,7 +932,7 @@ static inline void snd_pcm_gettime(struct snd_pcm_runtime *runtime,
        struct timespec *tv)
 {
  if (runtime->tstamp_type == SNDRV_PCM_TSTAMP_TYPE_MONOTONIC)
-  do_posix_clock_monotonic_gettime(tv);
+  ktime_get_ts(tv);
  else
   getnstimeofday(tv);
 }
diff --git a/sound/core/timer.c b/sound/core/timer.c
index 5bc1f3b..0aaf770 100644
--- a/sound/core/timer.c
+++ b/sound/core/timer.c
@@ -390,7 +390,7 @@ static void snd_timer_notify1(struct snd_timer_instance *ti, int event)
  struct timespec tstamp;
 
  if (timer_tstamp_monotonic)
-  do_posix_clock_monotonic_gettime(&tstamp);
+  ktime_get_ts(&tstamp);
  else
   getnstimeofday(&tstamp);
  if (snd_BUG_ON(event < SNDRV_TIMER_EVENT_START ||
@@ -1203,7 +1203,7 @@ static void snd_timer_user_tinterrupt(struct snd_timer_instance *timeri,
  }
  if (tu->last_resolution != resolution || ticks > 0) {
   if (timer_tstamp_monotonic)
-   do_posix_clock_monotonic_gettime(&tstamp);
+   ktime_get_ts(&tstamp);
   else
    getnstimeofday(&tstamp);
  }
-- 
1.7.1